patches/chromium/fix_add_support_for_webauthn_on_macos.patch

```diff
@@ -0,0 +1,315 @@
++  const PrefService* prefs = profile->prefs();
+   const base::ListValue* permit_attestation =
+       prefs->GetList(prefs::kSecurityKeyPermitAttestation);
+   return std::any_of(permit_attestation->GetList().begin(),
+@@ -236,16 +238,16 @@ bool ChromeWebAuthenticationDelegate::IsFocused(
+ // static
+ ChromeWebAuthenticationDelegate::TouchIdAuthenticatorConfig
+ ChromeWebAuthenticationDelegate::TouchIdAuthenticatorConfigForProfile(
+-    Profile* profile) {
++    ElectronBrowserContext* profile) {
+   constexpr char kKeychainAccessGroup[] =
+       MAC_TEAM_IDENTIFIER_STRING "." MAC_BUNDLE_IDENTIFIER_STRING ".webauthn";
+ 
+   std::string metadata_secret =
+-      profile->GetPrefs()->GetString(kWebAuthnTouchIdMetadataSecretPrefName);
++      profile->prefs()->GetString(kWebAuthnTouchIdMetadataSecretPrefName);
+   if (metadata_secret.empty() ||
+       !base::Base64Decode(metadata_secret, &metadata_secret)) {
+     metadata_secret = device::fido::mac::GenerateCredentialMetadataSecret();
+-    profile->GetPrefs()->SetString(
++    profile->prefs()->SetString(
+         kWebAuthnTouchIdMetadataSecretPrefName,
+         base::Base64Encode(base::as_bytes(base::make_span(metadata_secret))));
+   }
+@@ -259,7 +261,7 @@ absl::optional<ChromeWebAuthenticationDelegate::TouchIdAuthenticatorConfig>
+ ChromeWebAuthenticationDelegate::GetTouchIdAuthenticatorConfig(
+     content::BrowserContext* browser_context) {
+   return TouchIdAuthenticatorConfigForProfile(
+-      Profile::FromBrowserContext(browser_context));
++      static_cast<ElectronBrowserContext*>(browser_context));
+ }
+ #endif  // defined(OS_MAC)
+ 
+@@ -294,6 +296,7 @@ absl::optional<bool> ChromeWebAuthenticationDelegate::
+   }
+ #endif
+ 
++#if 0
+   // Chrome disables platform authenticators is Guest sessions. They may be
+   // available (behind an additional interstitial) in Incognito mode.
+   Profile* profile =
+@@ -301,6 +304,7 @@ absl::optional<bool> ChromeWebAuthenticationDelegate::
+   if (profile->IsGuestSession()) {
+     return false;
+   }
++#endif
+ 
+   return absl::nullopt;
+ }
+@@ -624,10 +628,12 @@ void ChromeAuthenticatorRequestDelegate::OnTransportAvailabilityEnumerated(
+ 
+   content::WebContents* web_contents =
+       content::WebContents::FromRenderFrameHost(GetRenderFrameHost());
++#if 0
+   Browser* browser = chrome::FindBrowserWithWebContents(web_contents);
+   if (browser) {
+     browser->window()->UpdatePageActionIcon(PageActionIconType::kWebAuthn);
+   }
++#endif
+ 
+   ShowAuthenticatorRequestDialog(web_contents,
+                                  std::move(transient_dialog_model_holder_));
+@@ -791,6 +797,7 @@ static std::string NameForDisplay(base::StringPiece raw_name) {
+   return base::UTF16ToUTF8(nonbreaking_name);
+ }
+ 
++#if 0
+ // PairingFromSyncedDevice extracts the caBLEv2 information from Sync's
+ // DeviceInfo (if any) into a caBLEv2 pairing. It may return nullptr.
```

Comment:
TIL https://blog.millerti.me/2021/06/18/previewing-chromes-cable-v2-support-for-webauthn/

obvs "cloud-assisted" isn't something we can support in Electron, but this is neat!